components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/resources/GroupResource.java

```diff
@@ -333,29 +462,144 @@ private Response processRequest(final Map<String, String> requestAttributes) {
+                            .toArray(new String[0]), permissionMap.get(SCIMProviderConstants.REMOVE)
+                            .toArray(new String[0]));
+
+                    outputPermissions = new JSONArray(Arrays.asList(userManager.getGroupPermissions(groupName)));
+                    scimResponse = new SCIMResponse(SCIMProviderConstants.HTTP_STATUS_SUCCESS, outputPermissions
+                            .toString(), responseHeaders);
+
+                } catch (JSONException e) {
+                    return SupportUtils.buildResponse(groupResourceManager.get(id, userManager, attributes,
+                            excludedAttributes));
+                }
             } else if (PATCH.class.getSimpleName().equals(httpVerb)) {
                 scimResponse = groupResourceManager
                         .updateWithPATCH(id, resourceString, userManager, attributes, excludedAttributes);
             } else if (DELETE.class.getSimpleName().equals(httpVerb)) {
                 scimResponse = groupResourceManager.delete(id, userManager);
             }
-            return SupportUtils.buildResponse(scimResponse);
+            return SupportUtils.buildResponse(Objects.requireNonNull(scimResponse));
+        } catch (BadRequestException e) {
+            logger.error("The Patch request is invalid. Unable to decode.");
+            return SupportUtils.buildResponse(new SCIMResponse(SCIMProviderConstants.HTTP_STATUS_BAD_REQUEST,
+                    "The Patch request is invalid.", responseHeaders));
         } catch (CharonException e) {
             return handleCharonException(e, encoder);
+        } catch (UserStoreException | RolePermissionException e) {
+            return handleCharonException(new CharonException("Error occurred when getting the permissions from server",
+                    e), encoder);
+        }
+    }
+
+    /**
+     * Get group display name from the groupId.
+     *
+     * @param groupId SCIM group id.
+     * @return Display name of the group.
+     * @throws JSONException thrown when an error occurred when getting displayName from JSON response.
+     */
+    private String getGroupName(String groupId, SCIMUserManager userManager, GroupResourceManager groupResourceManager,
+                                String excludeAttributes) throws JSONException {
+
+        String includeAttributes = SCIMConstants.GroupSchemaConstants.DISPLAY_NAME;
+        SCIMResponse scimResponse = groupResourceManager.get(groupId, userManager, includeAttributes, excludeAttributes);
+        JSONObject responseMessage = new JSONObject(scimResponse.getResponseMessage());
+        try {
+            return (String) (responseMessage).get(SCIMConstants.GroupSchemaConstants.DISPLAY_NAME);
+        } catch (JSONException e) {
+            throw new JSONException("An error occurred when extracting DisplayName from group: " + groupId);
```

Comment:
This hides the exact error **e**

Don't need to catch JSONException since it is thrown..